internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Analyze.scala

```diff
@@ -55,9 +52,9 @@ private[sbt] object Analyze {
         .orElse(Option(loadedClass.getEnclosingClass).flatMap(binaryToSourceName))
 
       srcClassName match {
-        case Some(srcClassName) =>
-          analysis.generatedNonLocalClass(source, newClass, binaryClassName, srcClassName)
-          classNames += srcClassName
+        case Some(className) =>
+          analysis.generatedNonLocalClass(source, newClass, binaryClassName, className)
+          classNames += className
```

Comment:
One of the wonderful things about pattern matching (in my opinion) is the ability to shadow.

Was this change triggered by a warning? If so, what was the warning about? And who emitted it?